pepper-apis/src/main/java/org/broadinstitute/ddp/route/UserRegistrationRoute.java

```diff
@@ -119,15 +124,50 @@ public Object handle(Request request, Response response, UserRegistrationPayload
             if (doLocalRegistration) {
                 auth0RefreshResponse = auth0Util.getRefreshTokenFromCode(payload.getAuth0Code(), auth0ClientId,
                         auth0ClientSecret, payload.getRedirectUri());
-                auth0UserId = Auth0Util.getVerifiedAuth0UserId(auth0RefreshResponse.getIdToken(), auth0Domain);
+                auth0UserId.set(Auth0Util.getVerifiedAuth0UserId(auth0RefreshResponse.getIdToken(), auth0Domain));
                 LOG.info("Successfully exchanged auth0 code for auth0UserId {} for local registration", auth0UserId);
             }
 
-            String ddpUserGuid;
-            User operatorUser = handle.attach(UserDao.class).findUserByAuth0UserId(auth0UserId, study.getAuth0TenantId()).orElse(null);
-            if (operatorUser == null) {
-                LOG.info("Attempting to register new user {} with client {} and study {}", auth0UserId, auth0ClientId, studyGuid);
-                operatorUser = registerUser(response, payload, handle, auth0Domain, auth0ClientId, auth0UserId);
+            String ddpUserGuid = null;
+            var userDao = handle.attach(UserDao.class);
+            User operatorUser = userDao.findUserByAuth0UserId(auth0UserId.get(), study.getAuth0TenantId()).orElse(null);
```

Comment:
Is there a reason for explicitly using `User` type instead of `var` here? I noticed there are a couple cases of this in the PR.